src/decode.rs

```diff
@@ -5105,6 +5096,7 @@ pub unsafe fn rav1d_submit_frame(c: &mut Rav1dContext) -> Rav1dResult {
                 for j in 0..i {
                     rav1d_thread_picture_unref(&mut f.refp[j]);
                 }
+                drop(task_thread_lock);
```

Comment:
This isn't needed anymore now that you've fixed `fn on_error` to not take `c: &mut` anymore, right?